diff -Naur mjpegtools-2.1.0.old/y4mutils/Makefile.am mjpegtools-2.1.0/y4mutils/Makefile.am
--- mjpegtools-2.1.0.old/y4mutils/Makefile.am	2020-04-03 15:17:34.622562677 +0200
+++ mjpegtools-2.1.0/y4mutils/Makefile.am	2020-04-03 15:17:45.040953129 +0200
@@ -44,6 +44,7 @@
 
 y4mhist_SOURCES = y4mhist.c
 y4mhist_LDADD = $(SDL_LIBS) $(SDLgfx_LIBS) $(LIBMJPEGUTILS)
+y4mhist_CPPFLAGS = $(AM_CPPFLAGS) $(SDL_CFLAGS)
 
 y4mblack_SOURCES = y4mblack.c
 y4mblack_LDADD = $(LIBMJPEGUTILS)
