diff -Naur mjpegtools-2.1.0.old/lavtools/jpeg2yuv.c mjpegtools-2.1.0/lavtools/jpeg2yuv.c
--- mjpegtools-2.1.0.old/lavtools/jpeg2yuv.c	2020-04-03 15:17:34.619562564 +0200
+++ mjpegtools-2.1.0/lavtools/jpeg2yuv.c	2020-04-03 15:18:18.149193939 +0200
@@ -451,7 +451,7 @@
            
            if (firstjpeg) {
                p = firstjpeg;
-               sprintf(jpegname, firstjpeg);
+               sprintf(jpegname, "%s", firstjpeg);
                firstjpeg = NULL;
            }
            else {
diff -Naur mjpegtools-2.1.0.old/lavtools/lav_io.c mjpegtools-2.1.0/lavtools/lav_io.c
--- mjpegtools-2.1.0.old/lavtools/lav_io.c	2020-04-03 15:17:34.620562602 +0200
+++ mjpegtools-2.1.0/lavtools/lav_io.c	2020-04-03 15:18:18.074191128 +0200
@@ -926,7 +926,7 @@
             strcat(errmsg, ", quicktime");
 #endif
 	    strcat(errmsg, ").\n");
-            fprintf(stderr, errmsg);
+        fprintf(stderr, "%s", errmsg);
 	    free(lav_fd);
 	    internal_error = ERROR_FORMAT; /* Format not recognized */
 	    return 0;
diff -Naur mjpegtools-2.1.0.old/lavtools/lavtrans.c mjpegtools-2.1.0/lavtools/lavtrans.c
--- mjpegtools-2.1.0.old/lavtools/lavtrans.c	2020-04-03 15:17:34.621562639 +0200
+++ mjpegtools-2.1.0/lavtools/lavtrans.c	2020-04-03 15:18:18.149193939 +0200
@@ -275,7 +275,7 @@
    if (process_image_frame != -1)
    {
       nv = el_get_video_frame(vbuff, process_image_frame, &el);
-      sprintf(imgfname,outfile);
+      sprintf(imgfname, "%s", outfile);
       imgfd = fopen(imgfname,"wb");
       if (imgfd==0) system_error("opening image file","fopen");
       res = fwrite(vbuff,nv,1,imgfd);
diff -Naur mjpegtools-2.1.0.old/mplex/main.cpp mjpegtools-2.1.0/mplex/main.cpp
--- mjpegtools-2.1.0.old/mplex/main.cpp	2020-04-03 15:17:34.593561590 +0200
+++ mjpegtools-2.1.0/mplex/main.cpp	2020-04-03 15:18:18.149193939 +0200
@@ -614,7 +614,7 @@
                 if( flag_table[flag].longname != 0 )
                     message += sep;
             }
-            mjpeg_error( message.c_str() );
+            mjpeg_error( "%s", message.c_str() );
             return false;
         }
 
diff -Naur mjpegtools-2.1.0.old/yuvfilters/yuvfilters.h mjpegtools-2.1.0/yuvfilters/yuvfilters.h
--- mjpegtools-2.1.0.old/yuvfilters/yuvfilters.h	2020-04-03 15:17:34.618562527 +0200
+++ mjpegtools-2.1.0/yuvfilters/yuvfilters.h	2020-04-03 15:18:18.149193939 +0200
@@ -34,9 +34,9 @@
 
 #ifdef MJPEGTOOLS
 # define perror(MSG) mjpeg_error("%s: %s", MSG, strerror(errno))
-# define WERROR(MSG) mjpeg_error(MSG)
-# define WERRORL(MSG) mjpeg_error(MSG)
-# define WWARN(MSG) mjpeg_warn(MSG)
+# define WERROR(MSG) mjpeg_error("%s", MSG)
+# define WERRORL(MSG) mjpeg_error("%s", MSG)
+# define WWARN(MSG) mjpeg_warn("%s", MSG)
 #else
 # define WERROR(MSG) write(2, MSG "\n", sizeof MSG)
 # define WERRORL(MSG) do { write(2, MSG, strlen(MSG)); write(2, "\n", 1); } while (0)
diff -Naur mjpegtools-2.1.0.old/yuvfilters/yuvkineco.c mjpegtools-2.1.0/yuvfilters/yuvkineco.c
--- mjpegtools-2.1.0.old/yuvfilters/yuvkineco.c	2020-04-03 15:17:34.618562527 +0200
+++ mjpegtools-2.1.0/yuvfilters/yuvkineco.c	2020-04-03 15:18:18.149193939 +0200
@@ -48,7 +48,7 @@
     } else {
 #ifdef MJPEGTOOLS
       buf[n] = '\0';
-      mjpeg_debug(buf);
+      mjpeg_debug("%s", buf);
 #else
       fputs(buf, stderr);
 #endif
